Rocket.Chat/Controllers/Preferences/Profile/EditProfileTableViewController.swift

```diff
@@ -525,7 +525,15 @@ extension EditProfileTableViewController: UIImagePickerControllerDelegate {
         let filename = String.random()
         var file: FileUpload?
 
-        if let image = info[.originalImage] as? UIImage {
+        if let image = info[.editedImage] as? UIImage {
+            file = UploadHelper.file(
+                for: image.compressedForUpload,
+                name: "\(filename.components(separatedBy: ".").first ?? "image").jpeg",
+                mimeType: "image/jpeg"
+            )
+
+            avatarView.imageView.image = image
+        } else if let image = info[.originalImage] as? UIImage {
```

Comment:
@shloak17107 It looks like the code here is the same as being used with the original image, do you think we can optimize it to only have it once?

Reply:
You are right. We can remove the else-if-condition where we are checking for the original image. This is because whenever the user selects an image for their avatar from the photo library, the cropping tool is always called. So, there will always be an .editedImage. 👍